solrman/smstorage/inmemory_storage.go

```diff
@@ -102,19 +101,34 @@ func (s *InMemoryStorage) GetStationaryOrgList() ([]string, error) {
 	return nil, nil
 }
 
-func (s *InMemoryStorage) IsDisabled() (bool, string) {
+func (s *InMemoryStorage) IsDisabled() (bool, error) {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 
-	return s.disabled, s.disabledReason
+	return len(s.disabledReasons) > 0, nil
 }
 
-func (s *InMemoryStorage) SetDisabled(disabled bool, reason string) error {
+func (s *InMemoryStorage) GetDisabledReasons() (map[string]string, error) {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	return s.disabledReasons, nil
+
+}
+
+func (s *InMemoryStorage) AddDisabledReason(requestor string, reason string) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	s.disabledReasons[requestor] = reason
```

Comment:
do we care if same requester wants to disable twice?

Reply:
no, it'll just overwrite the reason (the reason is icing on top for ops, the requestor is the key information really so that isn't an issue). The only scenario that's even relevant for is the UI button toggle, which isn't actually possible since it's a toggle.